src/NServiceBus.Core/Performance/Metrics/MetricsExtensions.cs

```diff
@@ -0,0 +1,52 @@
+namespace NServiceBus.Performance.Metrics
+{
+    using System;
+    using System.Collections.Generic;
+    using Features;
+
+    /// <summary>
+    /// Extension methods to configure OpenTelemetry metrics.
+    /// </summary>
+    public static class MetricsExtensions
+    {
+        /// <summary>
+        /// Configures NServiceBus to report messaging metrics using OpenTelemetry.
+        /// </summary>
+        /// <param name="config">The <see cref="EndpointConfiguration" /> instance to apply the settings to.</param>
+        public static EndpointConfiguration EnableOpenTelemetryMetrics(this EndpointConfiguration config)
+        {
+            Guard.AgainstNull(nameof(config), config);
+            config.EnableFeature<MessagingMetricsFeature>();
+
+            return config;
+        }
+    }
+
+    static class Extensions
+    {
+        public static void ThrowIfSendOnly(this FeatureConfigurationContext context)
+        {
+            var isSendOnly = context.Settings.GetOrDefault<bool>("Endpoint.SendOnly");
+            if (isSendOnly)
+            {
+                throw new Exception("Metrics are not supported on send only endpoints.");
+            }
+        }
+
+        public static bool TryGetMessageType(this ReceivePipelineCompleted completed, out string processedMessageType)
```

Comment:
nitpick but should we name this "TryGetMessageTypes" instead to keep it clear that it might contain multiple message types in the string?